content/en/docs/architecture/images.md

```diff
@@ -0,0 +1,132 @@
+
+## CI Images VS. ART Images
+From the above introduction, for each component, the CI image and the ART image are built from the same `Dockerfile` (DPTP has automation to refresh the `Dockerfile` path and name in the `ci-operator`'s configuration if either of them is modified in `ocp-build-data`). They are the same image if the images in `From` directives of `Dockerfile` are replaced by the same images. Actually, it is the case. In the above example, 
+ART **ensures** that `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6` is the same as `golang` in the ART build and `openshift-enterprise-base` is also an OCP component. Moreover, building the image from `Dockerfile` on a developer's computer leads to the same image as well.
+
+## GoLang Version
+As in the above example, the OCP component `cluster-etcd-operator` uses `GoLang-1.15` to build the image. When the version is evolved, ART team will modify the `GoLang` version in `ocp-build-data`, e.g., `GoLang-1.16`. Then, DPTP automation will create pull requests when it detects that `Dockerfile` in the component repo uses a different `GoLang` version from what is expected by `ocp-build-data`:
+
+* a PR (`PR1`) to the component repo: modify `From` directives of `Dockerfile`. It is up to the owner of the component whether or not to merge the PR.
+
+* a PR (`PR2`) to the `openshift/release` repo: modify the `ci-operator`'s configuration. This PR is merged automatically.
+
+{{< alert title="Info" color="info" >}}
+To avoid a large number of jobs triggered at the same time from PRs created from OCP components up to `GoLang` version change, the automation creates `PR1`s in sequence. It usually takes 1 or 2 days to complete this process.
+{{< /alert >}}
+
+Depending on when `PR1`s are merged, ART could use different versions of `GoLang` to build images.
+
+* Before code freeze: if less than half of the components merge `PR1`s, ART uses the old version `GoLang-1.15`. Otherwise, the new version `GoLang-1.16` for every component that has merged `PR1` and the old version `GoLang-1.15` for every component that has not.
+* After code freeze: the new version `GoLang-1.16` for all components, no mather if their `PR1`s have merged. Note that for each component that has not yet merged its `PR1` at this point, there is a blocker bug for the coming OCP release.
+
+{{< alert title="Warning" color="warning" >}}
+When `PR1` is created, the owner of the component should check its jobs to see how they get along with the new `GoLang` version. If an OCP component cannot migrate to the new version for the coming OCP release, the owner has to communicate with the ART team for this exceptional case after getting the approval from the architecture team.
+{{< /alert >}}
+
+CI images are built more closed to what `Dockerfile` defines in the component repo. E.g., after `PR2` is merged, the `ci-operator`'s configuration for `cluster-etcd-operator` looks like the following:
```

Comment:
I would remove this section and just say something like:

As two pull requests are created to reflect changes in the Go toolchain, it is possible for them to merge in either order. The CI system will be self-consistent and correct in either case. When only one of the changes (to `FROM` entries in he component repository and to `ci-operator` configuration in `openshift/release`) has merged, container image builds in CI will use the correct (newer) Go toolchain, but will simply be less efficient than they might be.

Reply:
Updated with your comment. BUT,

I thought I understood (after talking to you in a gmeeting). Probably still not.

Is the old version used in this case?

In case of
- change to FROM entries in the component repository has NOT merged, and
- change to ci-operator configuration in openshift/release has merged

That means ci config has `1.16` (the new version) as inputs which attempts to replace `1.15` (the older version) in Dockerfile.
We use the inputs as `build.spec.source.images.as`.

```bash
$ oc explain build.spec.source.images.as
KIND:     Build
VERSION:  build.openshift.io/v1

FIELD:    as <[]string>

DESCRIPTION:
     A list of image names that this source will be used in place of during a
     multi-stage container image build. For instance, a Dockerfile that uses
     "COPY --from=nginx:latest" will first check for an image source that has
     "nginx:latest" in this field before attempting to pull directly. If the
     Dockerfile does not reference an image source it is ignored. This field and
     paths may both be set, in which case the contents will be used twice.
```

 That is "If the Dockerfile does not reference an image source it is ignored", right?
So CI images is built from `1.15` as defined in the Dockerfile in the repo.

Reply:
The above is the builder image: `Dockerfile` in component repo rules.

The base image is the opposite: ci-config rules
```
$ oc explain build.spec.strategy.dockerStrategy.from
KIND:     Build
VERSION:  build.openshift.io/v1

RESOURCE: from <Object>

DESCRIPTION:
     from is a reference to an DockerImage, ImageStreamTag, or ImageStreamImage
     which overrides the FROM image in the Dockerfile for the build. If the
     Dockerfile uses multi-stage builds, this will replace the image in the last
     FROM directive of the file.

     ObjectReference contains enough information to let you inspect or modify
     the referred object.
```

Reply:
Yes, that is correct - in this case we will have a replacement rule in `ci-operator` configuration that does nothing, since there is no matching entry in the `Dockerfile`.